asyml/forte
# Docstring fix data converter

This PR partially corrects #610. 


* It will fix all scripts under `/forte/forte/data/converter`
    * `forte/data/converter/converter.py`
    * `forte/data/converter/feature.py`

### Description of changes
Describe what are the changes, and how they solve the problem.

### Possible influences of this PR.
It fixes docstrings of data converter to have the following features
* valid hyperlinks.
* docstring in major functions are shown correctly
* classes in `data.rst` includes all classes that needs documentation.
* correct indentation.

### Test Conducted
Check process
1. go through classes in the corresponding folders and check if it exists in sphinx documentation
2. use automation tool to check invalid hyperlinks
3. check indentation issues by visually going through sphinx documentation. 



### What's fixed?
* `LinkExtractor` and `SubwordExtractor` wasn't in sphinx configuration file. There were also some indentation issues in `default_configs()`.  Screenshots below shows effects after fixing.
![image](https://user-images.githubusercontent.com/28021889/152482398-7d079890-67f9-4f0c-91e3-3a61f5f2abcc.png)
![image](https://user-images.githubusercontent.com/28021889/152482662-af832eaa-82b1-42ca-ba00-d63cbb2b7b58.png)

* Two `Note` weren't aligned on a same indentation level initially. The screenshot below shows effects after fixing.
![image](https://user-images.githubusercontent.com/28021889/151897442-f7ecdbdb-dbb6-49ca-bc76-28811b466686.png)

* Fixed hyperlink of `BaseExtractor` and indentation for `dim`, `dtype`. Screenshot below shows effects after fixing.



![image](https://user-images.githubusercontent.com/28021889/152481235-32cd603d-6525-4416-bee6-8667b398f1dc.png)
